python/pyspark/mllib/feature.py

```diff
@@ -305,7 +350,12 @@ def fit(self, data):
                      treated as categorical for each distinct value.
                      Apply feature discretizer before using this function.
         """
-        jmodel = callMLlibFunc("fitChiSqSelector", self.numTopFeatures, data)
+        if self.selectorType == ChiSqSelectorType.KBest:
+            jmodel = callMLlibFunc("fitChiSqSelectorKBest", self.numTopFeatures, data)
+        elif self.selectorType == ChiSqSelectorType.Percentile:
+            jmodel = callMLlibFunc("fitChiSqSelectorPercentile", self.percentile, data)
+        elif self.selectorType == ChiSqSelectorType.FPR:
```

Comment:
I'm not sure what the convention in on the Pyspark side for args checking. For example here it doesn't check that the type is one of the supported types. I guess follow the same convention you find elsewhere.